geomstats/information_geometry/gamma.py

```diff
@@ -0,0 +1,1198 @@
+                - kappa**3 * gs.polygamma(3, kappa)
+                - 2 * kappa**2 * gs.polygamma(2, kappa)
+                - 2 * kappa * gs.polygamma(1, kappa)
+                + 1
+            )
+            / (2 * (kappa**2 * gs.polygamma(1, kappa) - kappa) ** 2),
+            0.5
+            * (
+                kappa**4
+                * (
+                    gs.polygamma(1, kappa) * gs.polygamma(3, kappa)
+                    - gs.polygamma(2, kappa) ** 2
+                )
+                - kappa**3 * gs.polygamma(3, kappa)
+                - 2 * kappa**2 * gs.polygamma(2, kappa)
+                - 2 * kappa * gs.polygamma(1, kappa)
+                + 1
+            ),
+        )
+
+        jac = gs.array(
+            [
+                [
+                    [[term_5, term_0], [term_0, term_0]],
+                    [[term_0, term_0], [term_4, term_2]],
+                ],
+                [
+                    [[term_0, term_0], [term_3, term_0]],
+                    [[term_3, term_0], [term_0, term_1]],
+                ],
+            ]
+        )
+
+        if n_points > 1:
+            jac = gs.transpose(jac, [4, 0, 1, 2, 3])
+
+        return gs.squeeze(jac)
+
+    def exp(
+        self,
+        tangent_vec,
+        base_point,
+        n_steps=N_STEPS,
+        step="euler",
+        solver="geomstats",
+    ):
+        """Compute the exponential map.
+
+        Comute the exponential map associated to the Fisher information metric
+        by solving the initial value problem associated to the geodesic
+        ordinary differential equation (ODE) using the Christoffel symbols.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., dim]
+            Base point.
+        n_steps : int
+            Number of steps for integration.
+            Optional, default: 100.
+        step : str, {'euler', 'rk4'}
+            Numerical scheme to use for integration.
+            Optional, default: 'euler'.
+        method : string, {'geomstats', 'lsoda'}
+            Method to compute the exponential map (self._exp_ivp or Connection.exp)
+            Optional, default : "geomstats"
+
+        Returns
+        -------
+        exp : array-like, shape=[..., dim]
+            End point of the geodesic starting at base_point with
+            initial velocity tangent_vec and stopping at time 1.
+        """
+        if solver == "geomstats":
+            return super(GammaMetric, self).exp(
+                tangent_vec, base_point, n_steps, step, solver
+            )
+        if solver == "lsoda":
+            return self._exp_ivp(tangent_vec, base_point, n_steps)
+
+    def log(
+        self,
+        point,
+        base_point,
+        n_steps=N_STEPS,
+        init=None,
+        step="euler",
+        max_iter=25,
+        verbose=False,
+        tol=gs.atol,
+        solver="geodesic_shooting",
```

Comment:
I think there was a mix up: here `method` makes more sense than `solver`, which refers to solving ODEs. See previous comment.